subsys/bluetooth/mesh/access.c

```diff
@@ -994,6 +1043,18 @@ int bt_mesh_comp_register(const struct bt_mesh_comp *comp)
 	return err;
 }
 
+int bt_mesh_comp2_register(const struct bt_mesh_comp2 *comp2)
+{
+	/* There must be at least one record */
+	if (!comp2 || !comp2->record_cnt) {
```

Comment:
We don't check input arguments, I think we can skip this error. We can mention in the API is that there must be at least one record in CDP2.